### DISCUSSION TITLE
useDim and cluster-specific p-values for pruneRegulon (using non-ArchR workflow)

### DISCUSSION THREAD

**Discussion by lmurrow** [xiaosaiyao/epiregulon#12] [General]
p2g takes a reducedDim matrix as an input (iterative LSI from the ATAC data), but  it seems for pruneRegulon the reducedDim matrix needs to be added as a reducedDim to the expMatrix (with reducedDimName as the inpute to "useDim")--see lines 36/37. If my understanding is correct here, would be worth clarifying in the tutorial or documentation.

For calculating cluster-specific p-values, might be worth adding a check to make sure a cluster has more cells than "cellNum" and remove any clusters that are too small from the analysis. This was an edge case that occurred for me based on subsetting to a particular cell type where one sample had very few cells.

**Comment by lmurrow** [comment]
Adding one more feature/check that would be convenient to include pruneRegulon. I ran into an error with duplicate rownames for my expMatrix (when named by gene symbols rather than ensemble ID, as reguired for matching with tf/target in regulon). Could be worth adding a stop error at the beginning of the code or just auto-renaming duplicates.

`Error in `levels<-`(`*tmp*`, value = as.character(levels)) : 
  factor level [2726] is duplicated
> traceback()
3: factor(regulon$target, levels = rownames(expMatrix))
2: countCells(regulon, expMatrix, peakMatrix, cluster_id, peak_cutoff, 
              exp_cutoff, clusters)
1: pruneRegulon(expMatrix = GeneExpressionMatrix, exp_assay = "logcounts", 
                peakMatrix = PeakMatrix_sub, peak_assay = "PeakMatrix", test = "chi.sq", 
                regulon_ChIP, clusters = GeneExpressionMatrix$PatientID.Visit, 
                prune_value = "pval", regulon_cutoff = 0.05, useDim = "IterativeLSI_ATAC", 
                cellNum = 10)`

**Reply by xiaosaiyao** [reply]
@TomVuod 

Thank you for your suggestions @lmurrow
